### PR TITLE
feat(jenkins): GH Checks for the PR

### PR DESCRIPTION
**Pull request by v1v**
## Highlights
- Create GH checks with the most important stages in the CI Pipeline.
- This particular change might help to debug when there are errors in the CI and identify them easily within then PR itself.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -260,27 +272,19 @@ pipeline {
...
-    aborted {
-      echoColor(text: '[ABORTED]', colorfg: 'magenta', colorbg: 'default')
-    }
-    failure {
-      echoColor(text: '[FAILURE]', colorfg: 'red', colorbg: 'default')
-      step([$class: 'Mailer', notifyEveryUnstableBuild: true, recipients: "${NOTIFY_TO}", sendToIndividuals: false])
-    }
-    unstable {
-      echoColor(text: '[UNSTABLE]', colorfg: 'yellow', colorbg: 'default')
+    always {
```
Comment (kuisathaverat) [inline Jenkinsfile]: if we merge this one we can close https://github.com/elastic/apm-agent-java/pull/655

Comment (v1v) [reply Jenkinsfile]: aha, I didn't see that PR, therefore I'll remove the buildNotify step in this PR as the other one seems to have some active conversation.

**Comment by mdelapenya** [comment]
why last revert?

**Comment by v1v** [comment]
There was another PR to change the notify step with some chat about it:
- https://github.com/elastic/apm-agent-java/pull/655

**Pull request merged**